### ISSUE TITLE
Doesn't merge in-line CSS / Style tag properly. Creates its own / new <style> section

### ISSUE THREAD

**Issue by Acecool** [htacg/tidy-html5#127]
When an existing <style type="text/css"> exists, instead of tidy adding the data to it, it creates its' own <style type="text/css">, and it doesn't index it properly either.

**Comment by geoffmcl** [comment]
Sorry for the delay in a reply...
We are now working on the latest development branch, develop-500
$ git pull and git checkout develop-500
Could you supply a small html sample that demonstrates this problem (in this dev branch)...

**Comment by balthisar** [comment]
Given the input
~~~~~
<!DOCTYPE html>
<html>
<head>
<style type="text/css">
p { font-size: 12pt; }
</style>
<title>Test</title>
</head>
<body>
<p style="color: red">Hello, world.</p>
<center>I'm in the middle</center>
</body>
</html>
~~~~~

I get the output
~~~~~
<!DOCTYPE html>
<html>
<head>

<style type="text/css">
p { font-size: 12pt; }
</style>
<title>Test</title>

<style type="text/css">
 div.c2 {text-align: center}
 p.c1 {color: red}
</style>
</head>
<body>
<p class="c1">Hello, world.</p>
<div class="c2">I'm in the middle</div>
</body>
</html>
~~~~~

I think what @Acecool is suggesting is that he would prefer to see the styles added to the existing `style` block in the `head`. I hope he comes back to confirm. In the meantime I will change this to a feature request and possible future milestone.

**Comment by geoffmcl** [comment]
While researching a fix for Issue #126 looked at why does tidy always create a new style section when doing a clean up, if this is the exact meaning of this issue.

At first it appeared to search for an existing style section, but the FindStyle(...) service it uses in clean.c ONLY searches for any previous style section it has created, and does not really do a full document search for any other style section.

So this feature request will have to wait for a bigger fix...

**Comment by geoffmcl** [comment]
I have done some more research into this, and while it would not be too difficult to do a full document search for any existing style block, I personally do not free this is in any way critical, so that change would not be by me.

Most certainly if someone presents a patch or a PR, it would be considered...

And just a reminder we are now using the `master` branch for development, and may eventually delete the develop-500 branch, although for the moment am trying to remember to keep it equal...

Since there have been no further comments on this issue since Feb 2015 will close it, but feel free to re-open, on present a PR against the current master.
